### PR TITLE
Shuffle some test initialization code to allow running under xdist.

### PR DESCRIPTION
**Pull request by kmaglione**
I was trying to get my tests to run under xdist in my `factory_boy` branch, and I wound up having to make these changes for the child process to initialize properly. That branch isn't ready for a PR yet, so I thought I may as well submit these separately.

### CODE REVIEW

**Comment by kmaglione** [comment]
Turns out this is actually fiendishly difficult to do correctly. My first version worked fine for me, but failed under Travis.

In the end, I think I found a solution that doesn't feel entirely hack-ish.

**Comment by magopian** [comment]
Excellent! I'm not sure about the import of `setup_olympia.py` as a django installed app, it's really hackish, and pretty sure it won't work anymore once we upgrade do django 1.7 and its not app loading code. Why don't we simply import this file in the `conftest.py` instead of the previously imported `manage.py`?

**Comment by magopian** [comment]
r+wc

**Comment by kmaglione** [comment]
I added comments, and also a couple of lines in `conftest.py` to also add prefixes to cache keys, just in case we do end up needing to switch back to using an actual memcache server.

File: conftest.py [hunk 1]
```diff
@@ -22,3 +14,33 @@ def mock_inline_css(monkeypatch):
     """
     import amo.helpers
     monkeypatch.setattr(amo.helpers, 'is_external', lambda css: True)
+
+
+def prefix_indexes(config):
+    """Prefix all ES index names and cache keys with `test_` and, if running
+    under xdist, the ID of the current slave."""
+
+    if hasattr(config, 'slaveinput'):
```
Comment (magopian) [inline conftest.py]: Can you add a comment about that? Explaining that it means we're running under xdist?

File: conftest.py [hunk 2]
```diff
@@ -22,3 +14,33 @@ def mock_inline_css(monkeypatch):
...
+    # Unfurtunately, it's surprisingly difficult to achieve with our current
+    # unittest-based setup.
+
+    for key, index in settings.ES_INDEXES.items():
+        if not index.startswith(prefix):
+            settings.ES_INDEXES[key] = '{prefix}_amo_{index}'.format(
+                prefix=prefix, index=index)
+
+    settings.CACHE_PREFIX = 'amo:{0}:'.format(prefix)
+    settings.KEY_PREFIX = settings.CACHE_PREFIX
```
Comment (magopian) [inline conftest.py]: I have the feeling this setting is unused (or rather, that it's just the same as `CACHE_PREFIX` everywhere, but that's something to be checked/fixed in another PR if needed ;)

**Comment by magopian** [comment]
One more comment and we're good to go, thanks!

r+wc

**Pull request merged**